src/object.h

```diff
@@ -28,4 +30,36 @@ int git_oid__parse(git_oid *oid, const char **buffer_out, const char *buffer_end
 
 void git_oid__writebuf(git_buf *buf, const char *header, const git_oid *oid);
 
+GIT_INLINE(bool) git_object__is_valid(
+	git_repository *repo, const git_oid *id, git_otype type)
+{
+	git_object *obj = NULL;
+	bool valid = true;
+
+	if (git_object__strict_input_validation) {
+		if (git_object_lookup(&obj, repo, id, type) < 0)
```

Comment:
Okay, I did indeed add `git_object__lookup_type` which will use the cache when possible and then go to `git_odb_read_header` when the requested object is not in the cache.  Thanks, @vmg !